services/google-data-api/src/index.js

```diff
@@ -0,0 +1,83 @@
+const { service } = require('@base-cms/micro');
+const { stringify } = require('querystring');
+
+const { name, version } = require('../package.json');
+const fetch = require('./fetch');
+const {
+  connect,
+  ping,
+  retrieve,
+  write,
+} = require('./mongodb');
+
+const { log } = console;
+
+const sortObject = (object = {}) => {
+  const keys = Object.keys(object).sort();
+  return keys.reduce((obj, key) => ({ ...obj, [key]: object[key] }), {});
+};
+
+module.exports = service.json({
+  init: async () => {
+    log(`> Booting ${name} ${version}...`);
+    log('> Connecting to MongoDB...');
+    await connect();
+  },
+  ping,
+  actions: {
```

Comment:
Move these actions to own folder/files...
`/actions/index.js` requires exports `/actions/youtube`
`/actions/youtube/index.js` requires and exports each individual action file, e.g. `actions/youtube/channel-list.js`

you can make a "load" helper that flattens these:
```js
const load = (root, obj) => keys(obj).reduce((o, key) => {
  const k = `${root}.${key}`;
  return { ...o, [k]: obj[key] };
}, {});
```

and then:
```js
module.exports = {
  ...load('youtube', youtube),
};
```

Alternatively, use `@base-cms/object-path` in `@base-cms/micro/service` so deep object retrieval "just works....

```js
const { json, createError } = require('micro');
const { get } = require('@base-cms/object-path');
const createParamError = require('./param-error');
const jsonErrors = require('./json-errors');

module.exports = async ({
  init,
  actions = {},
  ctx = {},
  healthPath = '/_health',
  ping,
  onError,
} = {}) => {
  if (typeof init === 'function') await init();
  return jsonErrors(async (req, res) => {
    const { url } = req;
    if (url === healthPath) {
      if (typeof ping === 'function') await ping(req);
      return { ok: true };
    }

    const input = await json(req);
    const { action: path, params, meta } = input;
    if (!path) throw createError(400, 'No action provided.');

    const fn = get(actions, path);
    if (typeof fn !== 'function') throw createParamError('action', path, Object.keys(actions));

    const data = await fn(params || {}, {
      ...ctx,
      req,
      res,
      meta: meta || {},
    });
    return { data };
  }, onError);
};
```